### PR TITLE
refactor(interactions): use timers/promises

### PR DESCRIPTION
**Pull request by DTrombett**
**Please describe the changes this PR makes and why it should be merged:**

Use the built-in promisified setTimeout `require('timers/promises').setTimeout` instead of `require('util').promisify(setTimeout)`

### CODE REVIEW

File: guide/interactions/buttons.md [hunk 1]
```diff
@@ -206,7 +206,7 @@ In addition to deferring an interaction response, you can defer the button, whic
 <!-- eslint-skip -->
 
 ```js {7-9}
-const wait = require('util').promisify(setTimeout);
+const wait = require('timers/promises').setTimeout;
```
Comment (almeidx) [inline guide/interactions/buttons.md:209]:
```suggestion
const wait = require('node:timers/promises').setTimeout;
```

File: guide/interactions/replying-to-slash-commands.md [hunk 2]
```diff
@@ -117,7 +117,7 @@ In this case, you can make use of the `CommandInteraction#deferReply()` method,
 <!--- here either display the is thinking message via vue-discord-message or place a screenshot -->
 
 ```js {7-9}
-const wait = require('util').promisify(setTimeout);
+const wait = require('timers/promises').setTimeout;
```
Comment (almeidx) [inline guide/interactions/replying-to-slash-commands.md:120]:
```suggestion
const wait = require('node:timers/promises').setTimeout;
```

File: guide/interactions/select-menus.md [hunk 3]
```diff
@@ -169,7 +169,7 @@ client.on('interactionCreate', async interaction => {
 Additionally to deferring the response of the interaction, you can defer the menu, which will trigger a loading state and then revert back to its original state:
 
 ```js {1,6-10}
-const wait = require('util').promisify(setTimeout);
+const wait = require('timers/promises').setTimeout;
```
Comment (almeidx) [inline guide/interactions/select-menus.md:172]:
```suggestion
const wait = require('node:timers/promises').setTimeout;
```

**Review by almeidx — CHANGES_REQUESTED** [review]

File: guide/interactions/buttons.md (hunk 1 above)
Comment (DTrombett) [reply guide/interactions/buttons.md:209]: I first added the `node:` protocol but then I saw that it's not used anywhere in the guide...
Not sure if it's ok if I add that here or if it's better to add all the `node:` protocols missing in another PR

Comment (Jiralite) [reply guide/interactions/buttons.md:209]: I'd say if there are more changes needed for `node:` outside of this pull request, do it separately.

**Review by almostSouji — CHANGES_REQUESTED** [review]
needs to be rebased after recent merges

**Comment by DTrombett** [comment]
> needs to be rebased after recent merges

Done!

**Pull request merged**